### PR TITLE
sbp-over-can: initial submit [DEVC-1200]

### PR DESCRIPTION
**Pull request by pmiettinen**
Ready for initial comments

### CODE REVIEW

File: board/piksiv3/linux.config [hunk 1]
```diff
@@ -61,8 +61,14 @@ CONFIG_IP_PNP_RARP=y
 CONFIG_NET_IPIP=m
 CONFIG_VLAN_8021Q=m
 CONFIG_CAN=y
+CONFIG_CAN_DEV=y
+CONFIG_CAN_CALC_BITTIMING=y
+CONFIG_CAN_RAW=y
 CONFIG_CAN_XILINXCAN=y
 CONFIG_CAN_DEBUG_DEVICES=y
+CONFIG_CAN_C_CAN_PLATFORM=y
```
Comment (pmiettinen) [inline board/piksiv3/linux.config]: These are just lucky-tries to get the `can_*` stuff to work, most likely unnecessary.

**Comment by anonymous-travis-swiftnav** [comment]
## HITL smoke tests: v2.0.0-develop-2018101122-3-g971da93e
These test runs are kicked off whenever you push a new commit to this PR. All passfail metrics in these runs must pass for the `hitl/pass-fail` status to be marked successful.
### job status
+ [live-roof-650-townsend-post runs](https://gnss-analysis.swiftnav.com/jobs/id=717990)
### gnss-analysis results
At least one run must complete for these links to have data.
#### passfail
+ [live-roof-650-townsend-post](https://gnss-analysis.swiftnav.com/summary_type=q50&metrics_preset=pass_fail&scenario=live-roof-650-townsend-post&build_type=buildroot_pr&firmware_versions=v2.0.0-develop-2018101122-3-g971da93e&groupby_key=firmware&display_type=table)
#### detailed
+ [live-roof-650-townsend-post](https://gnss-analysis.swiftnav.com/summary_type=q50&metrics_preset=detailed&scenario=live-roof-650-townsend-post&build_type=buildroot_pr&firmware_versions=v2.0.0-develop-2018101122-3-g971da93e&groupby_key=firmware&display_type=table)

**Comment by anonymous-travis-swiftnav** [comment]
## v2.0.0-develop-2018101122-3-g971da93e
+ [Artifacts (HITL Dashboard Page)](https://gnss-analysis.swiftnav.com/artifacts/build_type=buildroot_pr&name_filter=v2.0.0-develop-2018101122-3-g971da93e)
+ [Artifacts (S3 / AWS Console)](https://console.aws.amazon.com/s3/home?region=us-west-2&bucket=swiftnav-artifacts-pull-requests&prefix=piksi_buildroot/v2.0.0-develop-2018101122-3-g971da93e/)
+ s3://swiftnav-artifacts-pull-requests/piksi_buildroot/v2.0.0-develop-2018101122-3-g971da93e

**Comment by anonymous-travis-swiftnav** [comment]
## HITL smoke tests: v2.0.0-develop-2018101616-2-gf00ba7af
These test runs are kicked off whenever you push a new commit to this PR. All passfail metrics in these runs must pass for the `hitl/pass-fail` status to be marked successful.
### job status
+ [live-roof-650-townsend-post runs](https://gnss-analysis.swiftnav.com/jobs/id=724994)
### gnss-analysis results
At least one run must complete for these links to have data.
#### passfail
+ [live-roof-650-townsend-post](https://gnss-analysis.swiftnav.com/summary_type=q50&metrics_preset=pass_fail&scenario=live-roof-650-townsend-post&build_type=buildroot_pr&firmware_versions=v2.0.0-develop-2018101616-2-gf00ba7af&groupby_key=firmware&display_type=table)
#### detailed
+ [live-roof-650-townsend-post](https://gnss-analysis.swiftnav.com/summary_type=q50&metrics_preset=detailed&scenario=live-roof-650-townsend-post&build_type=buildroot_pr&firmware_versions=v2.0.0-develop-2018101616-2-gf00ba7af&groupby_key=firmware&display_type=table)

**Comment by anonymous-travis-swiftnav** [comment]
## v2.0.0-develop-2018101616-2-gf00ba7af
+ [Artifacts (HITL Dashboard Page)](https://gnss-analysis.swiftnav.com/artifacts/build_type=buildroot_pr&name_filter=v2.0.0-develop-2018101616-2-gf00ba7af)
+ [Artifacts (S3 / AWS Console)](https://console.aws.amazon.com/s3/home?region=us-west-2&bucket=swiftnav-artifacts-pull-requests&prefix=piksi_buildroot/v2.0.0-develop-2018101616-2-gf00ba7af/)
+ s3://swiftnav-artifacts-pull-requests/piksi_buildroot/v2.0.0-develop-2018101616-2-gf00ba7af

**Comment by anonymous-travis-swiftnav** [comment]
## HITL smoke tests: v2.0.0-develop-2018101616-3-gd6ff8ce1
These test runs are kicked off whenever you push a new commit to this PR. All passfail metrics in these runs must pass for the `hitl/pass-fail` status to be marked successful.
### job status
+ [live-roof-650-townsend-post runs](https://gnss-analysis.swiftnav.com/jobs/id=725231)
### gnss-analysis results
At least one run must complete for these links to have data.
#### passfail
+ [live-roof-650-townsend-post](https://gnss-analysis.swiftnav.com/summary_type=q50&metrics_preset=pass_fail&scenario=live-roof-650-townsend-post&build_type=buildroot_pr&firmware_versions=v2.0.0-develop-2018101616-3-gd6ff8ce1&groupby_key=firmware&display_type=table)
#### detailed
+ [live-roof-650-townsend-post](https://gnss-analysis.swiftnav.com/summary_type=q50&metrics_preset=detailed&scenario=live-roof-650-townsend-post&build_type=buildroot_pr&firmware_versions=v2.0.0-develop-2018101616-3-gd6ff8ce1&groupby_key=firmware&display_type=table)

**Comment by anonymous-travis-swiftnav** [comment]
## v2.0.0-develop-2018101616-3-gd6ff8ce1
+ [Artifacts (HITL Dashboard Page)](https://gnss-analysis.swiftnav.com/artifacts/build_type=buildroot_pr&name_filter=v2.0.0-develop-2018101616-3-gd6ff8ce1)
+ [Artifacts (S3 / AWS Console)](https://console.aws.amazon.com/s3/home?region=us-west-2&bucket=swiftnav-artifacts-pull-requests&prefix=piksi_buildroot/v2.0.0-develop-2018101616-3-gd6ff8ce1/)
+ s3://swiftnav-artifacts-pull-requests/piksi_buildroot/v2.0.0-develop-2018101616-3-gd6ff8ce1

**Comment by anonymous-travis-swiftnav** [comment]
## HITL smoke tests: v2.0.0-develop-2018101616-8-ga479d27a
These test runs are kicked off whenever you push a new commit to this PR. All passfail metrics in these runs must pass for the `hitl/pass-fail` status to be marked successful.
### job status
+ [live-roof-650-townsend-post runs](https://gnss-analysis.swiftnav.com/jobs/id=726767)
### gnss-analysis results
At least one run must complete for these links to have data.
#### passfail
+ [live-roof-650-townsend-post](https://gnss-analysis.swiftnav.com/summary_type=q50&metrics_preset=pass_fail&scenario=live-roof-650-townsend-post&build_type=buildroot_pr&firmware_versions=v2.0.0-develop-2018101616-8-ga479d27a&groupby_key=firmware&display_type=table)
#### detailed
+ [live-roof-650-townsend-post](https://gnss-analysis.swiftnav.com/summary_type=q50&metrics_preset=detailed&scenario=live-roof-650-townsend-post&build_type=buildroot_pr&firmware_versions=v2.0.0-develop-2018101616-8-ga479d27a&groupby_key=firmware&display_type=table)

**Comment by anonymous-travis-swiftnav** [comment]
## v2.0.0-develop-2018101616-8-ga479d27a
+ [Artifacts (HITL Dashboard Page)](https://gnss-analysis.swiftnav.com/artifacts/build_type=buildroot_pr&name_filter=v2.0.0-develop-2018101616-8-ga479d27a)
+ [Artifacts (S3 / AWS Console)](https://console.aws.amazon.com/s3/home?region=us-west-2&bucket=swiftnav-artifacts-pull-requests&prefix=piksi_buildroot/v2.0.0-develop-2018101616-8-ga479d27a/)
+ s3://swiftnav-artifacts-pull-requests/piksi_buildroot/v2.0.0-develop-2018101616-8-ga479d27a

**Comment by anonymous-travis-swiftnav** [comment]
## HITL smoke tests: v2.0.0-develop-2018101616-10-gd5424b18
These test runs are kicked off whenever you push a new commit to this PR. All passfail metrics in these runs must pass for the `hitl/pass-fail` status to be marked successful.
### job status
+ [live-roof-650-townsend-post runs](https://gnss-analysis.swiftnav.com/jobs/id=727681)
### gnss-analysis results
At least one run must complete for these links to have data.
#### passfail
+ [live-roof-650-townsend-post](https://gnss-analysis.swiftnav.com/summary_type=q50&metrics_preset=pass_fail&scenario=live-roof-650-townsend-post&build_type=buildroot_pr&firmware_versions=v2.0.0-develop-2018101616-10-gd5424b18&groupby_key=firmware&display_type=table)
#### detailed
+ [live-roof-650-townsend-post](https://gnss-analysis.swiftnav.com/summary_type=q50&metrics_preset=detailed&scenario=live-roof-650-townsend-post&build_type=buildroot_pr&firmware_versions=v2.0.0-develop-2018101616-10-gd5424b18&groupby_key=firmware&display_type=table)

**Comment by anonymous-travis-swiftnav** [comment]
## v2.0.0-develop-2018101616-10-gd5424b18
+ [Artifacts (HITL Dashboard Page)](https://gnss-analysis.swiftnav.com/artifacts/build_type=buildroot_pr&name_filter=v2.0.0-develop-2018101616-10-gd5424b18)
+ [Artifacts (S3 / AWS Console)](https://console.aws.amazon.com/s3/home?region=us-west-2&bucket=swiftnav-artifacts-pull-requests&prefix=piksi_buildroot/v2.0.0-develop-2018101616-10-gd5424b18/)
+ s3://swiftnav-artifacts-pull-requests/piksi_buildroot/v2.0.0-develop-2018101616-10-gd5424b18

**Comment by anonymous-travis-swiftnav** [comment]
## HITL smoke tests: v2.0.0-develop-2018101616-11-g8a0b42ce
These test runs are kicked off whenever you push a new commit to this PR. All passfail metrics in these runs must pass for the `hitl/pass-fail` status to be marked successful.
### job status
+ [live-roof-650-townsend-post runs](https://gnss-analysis.swiftnav.com/jobs/id=727831)
### gnss-analysis results
At least one run must complete for these links to have data.
#### passfail
+ [live-roof-650-townsend-post](https://gnss-analysis.swiftnav.com/summary_type=q50&metrics_preset=pass_fail&scenario=live-roof-650-townsend-post&build_type=buildroot_pr&firmware_versions=v2.0.0-develop-2018101616-11-g8a0b42ce&groupby_key=firmware&display_type=table)
#### detailed
+ [live-roof-650-townsend-post](https://gnss-analysis.swiftnav.com/summary_type=q50&metrics_preset=detailed&scenario=live-roof-650-townsend-post&build_type=buildroot_pr&firmware_versions=v2.0.0-develop-2018101616-11-g8a0b42ce&groupby_key=firmware&display_type=table)

**Comment by anonymous-travis-swiftnav** [comment]
## v2.0.0-develop-2018101616-11-g8a0b42ce
+ [Artifacts (HITL Dashboard Page)](https://gnss-analysis.swiftnav.com/artifacts/build_type=buildroot_pr&name_filter=v2.0.0-develop-2018101616-11-g8a0b42ce)
+ [Artifacts (S3 / AWS Console)](https://console.aws.amazon.com/s3/home?region=us-west-2&bucket=swiftnav-artifacts-pull-requests&prefix=piksi_buildroot/v2.0.0-develop-2018101616-11-g8a0b42ce/)
+ s3://swiftnav-artifacts-pull-requests/piksi_buildroot/v2.0.0-develop-2018101616-11-g8a0b42ce

**Comment by benjaminaltieri** [comment]
I'll push a quick `clang-format` to get this building and rebased on current master

**Comment by anonymous-travis-swiftnav** [comment]
## HITL smoke tests: v2.0.0-develop-2018102218-21-g862cac78
These test runs are kicked off whenever you push a new commit to this PR. All passfail metrics in these runs must pass for the `hitl/pass-fail` status to be marked successful.
### job status
+ [live-roof-650-townsend-post runs](https://gnss-analysis.swiftnav.com/jobs/id=739700)
### gnss-analysis results
At least one run must complete for these links to have data.
#### passfail
+ [live-roof-650-townsend-post](https://gnss-analysis.swiftnav.com/summary_type=q50&metrics_preset=pass_fail&scenario=live-roof-650-townsend-post&build_type=buildroot_pr&firmware_versions=v2.0.0-develop-2018102218-21-g862cac78&groupby_key=firmware&display_type=table)
#### detailed
+ [live-roof-650-townsend-post](https://gnss-analysis.swiftnav.com/summary_type=q50&metrics_preset=detailed&scenario=live-roof-650-townsend-post&build_type=buildroot_pr&firmware_versions=v2.0.0-develop-2018102218-21-g862cac78&groupby_key=firmware&display_type=table)

**Comment by anonymous-travis-swiftnav** [comment]
## v2.0.0-develop-2018102218-21-g862cac78
+ [Artifacts (HITL Dashboard Page)](https://gnss-analysis.swiftnav.com/artifacts/build_type=buildroot_pr&name_filter=v2.0.0-develop-2018102218-21-g862cac78)
+ [Artifacts (S3 / AWS Console)](https://console.aws.amazon.com/s3/home?region=us-west-2&bucket=swiftnav-artifacts-pull-requests&prefix=piksi_buildroot/v2.0.0-develop-2018102218-21-g862cac78/)
+ s3://swiftnav-artifacts-pull-requests/piksi_buildroot/v2.0.0-develop-2018102218-21-g862cac78

File: package/ports_daemon/ports_daemon/src/ports.c [hunk 2]
```diff
@@ -266,6 +282,30 @@ static port_config_t port_configs[] = {
     .restart = DO_NOT_RESTART,
     .first_start = true,
   },
+  {
```
Comment (silverjam) [inline package/ports_daemon/ports_daemon/src/ports.c]: @benjaminaltieri We discussed making this a dev only change by disabling can0/can1 except when `/persistent/flags/enable_can` is `1`.   We also need to not add `can1` if we're on a `Duro`.

Comment (silverjam) [reply package/ports_daemon/ports_daemon/src/ports.c]: I've added these follow up tasks to cover this:
  - https://swift-nav.atlassian.net/browse/DEVC-1246
  - https://swift-nav.atlassian.net/browse/DEVC-1247

File: package/sbp_protocol/src/sbp_router.yml [hunk 3]
```diff
@@ -208,3 +217,7 @@ ports:
   - name: SBP_PORT_NMEA_BRIDGE
     pub_addr: "ipc:///var/run/sockets/nmea_bridge.pub"
     sub_addr: "ipc:///var/run/sockets/nmea_bridge.sub"
+
+  - name: SBP_PORT_CAN
```
[the remark this answers is not in the document]
Comment (pmiettinen) [reply package/sbp_protocol/src/sbp_router.yml]: I believe these can be removed.

File: board/piksiv3/linux.config (hunk 1 above)
Comment (pmiettinen) [reply board/piksiv3/linux.config]: Most of these are most likely not needed. I need someone with access to debug devices to test which ones are necessary.

**Comment by anonymous-travis-swiftnav** [comment]
## HITL smoke tests: v2.0.0-develop-2018102218-25-g65e1d174
These test runs are kicked off whenever you push a new commit to this PR. All passfail metrics in these runs must pass for the `hitl/pass-fail` status to be marked successful.
### job status
+ [live-roof-650-townsend-post runs](https://gnss-analysis.swiftnav.com/jobs/id=744836)
### gnss-analysis results
At least one run must complete for these links to have data.
#### passfail
+ [live-roof-650-townsend-post](https://gnss-analysis.swiftnav.com/summary_type=q50&metrics_preset=pass_fail&scenario=live-roof-650-townsend-post&build_type=buildroot_pr&firmware_versions=v2.0.0-develop-2018102218-25-g65e1d174&groupby_key=firmware&display_type=table)
#### detailed
+ [live-roof-650-townsend-post](https://gnss-analysis.swiftnav.com/summary_type=q50&metrics_preset=detailed&scenario=live-roof-650-townsend-post&build_type=buildroot_pr&firmware_versions=v2.0.0-develop-2018102218-25-g65e1d174&groupby_key=firmware&display_type=table)

**Comment by anonymous-travis-swiftnav** [comment]
## v2.0.0-develop-2018102218-25-g65e1d174
+ [Artifacts (HITL Dashboard Page)](https://gnss-analysis.swiftnav.com/artifacts/build_type=buildroot_pr&name_filter=v2.0.0-develop-2018102218-25-g65e1d174)
+ [Artifacts (S3 / AWS Console)](https://console.aws.amazon.com/s3/home?region=us-west-2&bucket=swiftnav-artifacts-pull-requests&prefix=piksi_buildroot/v2.0.0-develop-2018102218-25-g65e1d174/)
+ s3://swiftnav-artifacts-pull-requests/piksi_buildroot/v2.0.0-develop-2018102218-25-g65e1d174

**Pull request merged**